addon/services/notification-messages-service.js

```diff
@@ -1,127 +1,133 @@
-        // If no message is set, throw an error
-        if (!options.message) {
-            throw new Error("No notification message set");
-        }
-
-        const notification = Ember.Object.create({
-            message: options.message,
-            type: options.type || 'info', // info, success, warning, error
-            autoClear: (Ember.isEmpty(options.autoClear) ? this.get('defaultAutoClear') : options.autoClear),
-            clearDuration: options.clearDuration || this.get('defaultClearDuration'),
-            onClick: options.onClick,
-            htmlContent: options.htmlContent || false,
-            cssClasses: options.cssClasses
-        });
-
-        this.pushObject(notification);
-
-        if (notification.autoClear) {
-            notification.set('remaining', notification.get('clearDuration'));
-            this.setupAutoClear(notification);
-        }
-
-        return notification;
-    },
-
-    // Helper methods for each type of notification
-    error(message, options) {
-      return this.addNotification(assign({
-        message: message,
-        type: 'error'
-      }, options));
-    },
-
-    success(message, options) {
-      return this.addNotification(assign({
-        message: message,
-        type: 'success'
-      }, options));
-    },
-
-    info(message, options) {
-      return this.addNotification(assign({
-        message: message,
-        type: 'info'
-      }, options));
-    },
-
-    warning(message, options) {
-      return this.addNotification(assign({
-        message: message,
-        type: 'warning'
-      }, options));
-    },
-
-    removeNotification(notification) {
-        if (!notification) {
-            return;
-        }
-        notification.set('dismiss', true);
-        // Delay removal from DOM for dismissal animation
-        Ember.run.later(this, () => {
-            this.removeObject(notification);
-        }, 500);
-    },
-
-    setupAutoClear(notification) {
-        notification.set('startTime', Date.now());
-
-        const timer = Ember.run.later(this, () => {
-            // Hasn't been closed manually
-            if (this.indexOf(notification) >= 0) {
-                this.removeNotification(notification);
-            }
-        }, notification.get('remaining'));
-
-        notification.set('timer', timer);
-    },
-
-    pauseAutoClear(notification) {
-        Ember.run.cancel(notification.get('timer'));
-
-        const elapsed = Date.now() - notification.get('startTime');
-        const remaining = notification.get('clearDuration') - elapsed;
-        notification.set('remaining', remaining);
-    },
-
-    clearAll() {
-      this.get('content').forEach(notification => {
-          this.removeNotification(notification);
-      });
+    const notification = Object.create({
+      message: options.message,
+      type: options.type || 'info',
+      autoClear: (isEmpty(options.autoClear) ? this.get('globalAutoClear') : options.autoClear),
```

Comment:
because `this.get('globalAutoClear')` is only used exactly once I wouldn't complicate this and set it as a property on the NotificationMessagesService.

In fact, I don't understand the reasoning behind the `setGlobal` anyway 🤔 

You could simplify this a bit and just use "get with default" `Ember.getWithDefault(globals, 'autoClear', 'false');`